vcpkg's fix-deps.patch (ports/libarchive at 3.8.9), carried to 3.8.10 by
dosbox-automation: three added lines moved with 3.8.10's CMakeLists (bcrypt,
xmllite, expat); the added and removed lines are otherwise vcpkg's.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -501,6 +501,7 @@
   SET(HAVE_ZLIB_H 1)
   INCLUDE_DIRECTORIES(${ZLIB_INCLUDE_DIR})
   LIST(APPEND ADDITIONAL_LIBS ${ZLIB_LIBRARIES})
+  STRING(APPEND LIBSREQUIRED " zlib")
   IF(WIN32 AND NOT CYGWIN)
     #
     # Test if ZLIB_WINAPI macro is needed to use.
@@ -540,6 +541,7 @@
   SET(HAVE_BZLIB_H 1)
   INCLUDE_DIRECTORIES(${BZIP2_INCLUDE_DIR})
   LIST(APPEND ADDITIONAL_LIBS ${BZIP2_LIBRARIES})
+  STRING(APPEND LIBSREQUIRED " bzip2")
   # Test if a macro is needed for the library.
   TRY_MACRO_FOR_LIBRARY(
     "${BZIP2_INCLUDE_DIR}" "${BZIP2_LIBRARIES}"
@@ -568,6 +570,8 @@
 IF(LIBLZMA_FOUND)
   SET(HAVE_LIBLZMA 1)
   SET(HAVE_LZMA_H 1)
+  LIST(APPEND ADDITIONAL_LIBS LibLZMA::LibLZMA)
+  STRING(APPEND LIBSREQUIRED " liblzma")
   CMAKE_PUSH_CHECK_STATE()
   SET(CMAKE_REQUIRED_INCLUDES ${LIBLZMA_INCLUDE_DIR})
   SET(CMAKE_REQUIRED_LIBRARIES ${LIBLZMA_LIBRARIES})
@@ -615,6 +619,7 @@
   SET(HAVE_LZO_LZO1X_H 1)
   INCLUDE_DIRECTORIES(${LZO2_INCLUDE_DIR})
   LIST(APPEND ADDITIONAL_LIBS ${LZO2_LIBRARY})
+  STRING(APPEND LIBSREQUIRED " lzo2")
   #
   # TODO: test for static library.
   #
@@ -644,6 +649,7 @@
   SET(ARCHIVE_BLAKE2 FALSE)
   LIST(APPEND ADDITIONAL_LIBS ${LIBB2_LIBRARY})
   INCLUDE_DIRECTORIES(${LIBB2_INCLUDE_DIR})
+  LIST(APPEND LIBARCHIVE_LIBS_PRIVATE ${LIBB2_LIBRARY})
   CMAKE_PUSH_CHECK_STATE()
   SET(CMAKE_REQUIRED_LIBRARIES ${LIBB2_LIBRARY})
   SET(CMAKE_REQUIRED_INCLUDES ${LIBB2_INCLUDE_DIR})
@@ -661,6 +667,8 @@
 # Find LZ4
 #
 IF(ENABLE_LZ4)
+  FIND_PACKAGE(LZ4 NAMES lz4 CONFIG REQUIRED)
+elseif(0)
   IF (LZ4_INCLUDE_DIR)
     # Already in cache, be silent
     SET(LZ4_FIND_QUIETLY TRUE)
@@ -676,6 +684,10 @@
 IF(LZ4_FOUND)
   SET(HAVE_LIBLZ4 1)
   SET(HAVE_LZ4_H 1)
+  SET(HAVE_LZ4HC_H 1)
+  LIST(APPEND ADDITIONAL_LIBS lz4::lz4)
+  STRING(APPEND LIBSREQUIRED " liblz4")
+elseif(0)
   CMAKE_PUSH_CHECK_STATE()	# Save the state of the variables
   SET(CMAKE_REQUIRED_INCLUDES ${LZ4_INCLUDE_DIR})
   CHECK_INCLUDE_FILES("lz4hc.h" HAVE_LZ4HC_H)
@@ -692,6 +704,8 @@
 # Find Zstd
 #
 IF(ENABLE_ZSTD)
+  FIND_PACKAGE(ZSTD NAMES zstd CONFIG REQUIRED)
+elseif(0)
   IF (ZSTD_INCLUDE_DIR)
     # Already in cache, be silent
     SET(ZSTD_FIND_QUIETLY TRUE)
@@ -711,6 +725,11 @@
 ENDIF(ENABLE_ZSTD)
 IF(ZSTD_FOUND)
   SET(HAVE_ZSTD_H 1)
+  SET(HAVE_LIBZSTD 1)
+  SET(HAVE_ZSTD_compressStream 1)
+  LIST(APPEND ADDITIONAL_LIBS zstd::libzstd)
+  STRING(APPEND LIBSREQUIRED " libzstd")
+elseif(0)
   INCLUDE_DIRECTORIES(${ZSTD_INCLUDE_DIR})
   LIST(APPEND ADDITIONAL_LIBS ${ZSTD_LIBRARY})
   CMAKE_PUSH_CHECK_STATE()
@@ -824,6 +843,7 @@
 LA_CHECK_INCLUDE_FILE("bcrypt.h" HAVE_BCRYPT_H)
 IF(HAVE_BCRYPT_H)
   LIST(APPEND ADDITIONAL_LIBS "bcrypt")
+  LIST(APPEND LIBARCHIVE_LIBS_PRIVATE "bcrypt")
   IF(ENABLE_CNG)
     SET(ARCHIVE_CRYPTO_WIN        1)
     # bcrypt supports these algorithms on all available versions
@@ -865,6 +885,7 @@
   IF(MBEDTLS_FOUND)
     SET(HAVE_LIBMBEDCRYPTO 1)
     LIST(APPEND ADDITIONAL_LIBS ${MBEDCRYPTO_LIBRARY})
+    LIST(APPEND LIBARCHIVE_LIBS_PRIVATE ${MBEDCRYPTO_LIBRARY})
     INCLUDE_DIRECTORIES(${MBEDTLS_INCLUDE_DIRS})
 
     CMAKE_PUSH_CHECK_STATE()
@@ -888,6 +909,7 @@
   IF(NETTLE_FOUND)
     SET(HAVE_LIBNETTLE 1)
     LIST(APPEND ADDITIONAL_LIBS ${NETTLE_LIBRARIES})
+    STRING(APPEND LIBSREQUIRED " nettle")
     INCLUDE_DIRECTORIES(${NETTLE_INCLUDE_DIR})
 
     CMAKE_PUSH_CHECK_STATE()
@@ -916,8 +938,8 @@
   FIND_PACKAGE(OpenSSL)
   IF(OPENSSL_FOUND)
     SET(HAVE_LIBCRYPTO 1)
-    INCLUDE_DIRECTORIES(${OPENSSL_INCLUDE_DIR})
-    LIST(APPEND ADDITIONAL_LIBS ${OPENSSL_CRYPTO_LIBRARY})
+    LIST(APPEND ADDITIONAL_LIBS OpenSSL::Crypto)
+    STRING(APPEND LIBSREQUIRED " libcrypto")
 
     CMAKE_PUSH_CHECK_STATE()
     SET(CMAKE_REQUIRED_LIBRARIES ${OPENSSL_CRYPTO_LIBRARY})
@@ -940,6 +962,7 @@
     SET(CMAKE_REQUIRED_LIBRARIES "md")
     FIND_LIBRARY(LIBMD_LIBRARY NAMES md)
     LIST(APPEND ADDITIONAL_LIBS ${LIBMD_LIBRARY})
+    LIST(APPEND LIBARCHIVE_LIBS_PRIVATE ${LIBMD_LIBRARY})
     CMAKE_POP_CHECK_STATE()	# Restore the state of the variables
   ENDIF(LIBMD_FOUND)
 ENDIF(ENABLE_MD AND NOT OPENSSL_FOUND)
@@ -1057,7 +1080,7 @@
         IF ("${IMPLEMENTATION}" MATCHES "^OPENSSL$" AND OPENSSL_FOUND)
           INCLUDE_DIRECTORIES(${OPENSSL_INCLUDE_DIR})
 	  LIST(APPEND ADDITIONAL_LIBS ${OPENSSL_LIBRARIES})
-	   LIST(REMOVE_DUPLICATES ADDITIONAL_LIBS)
+           STRING(APPEND LIBSREQUIRED " libssl")
         ENDIF ("${IMPLEMENTATION}" MATCHES "^OPENSSL$" AND OPENSSL_FOUND)
       ENDIF (ARCHIVE_CRYPTO_${ALGORITHM}_${IMPLEMENTATION})
       ENDIF(NOT ARCHIVE_CRYPTO_${ALGORITHM})
@@ -1151,6 +1174,7 @@
       CHECK_ICONV("libiconv" "")
       IF (HAVE_ICONV)
         LIST(APPEND ADDITIONAL_LIBS ${LIBICONV_PATH})
+        LIST(APPEND LIBARCHIVE_LIBS_PRIVATE ${LIBICONV_PATH})
       ENDIF(HAVE_ICONV)
     ENDIF(NOT HAVE_ICONV AND LIBICONV_PATH)
   ENDIF(ICONV_INCLUDE_DIR)
@@ -1184,6 +1208,7 @@
       ENDIF(WIN32 AND NOT CYGWIN)
       IF(HAVE_LOCALE_CHARSET)
         LIST(APPEND ADDITIONAL_LIBS ${LIBCHARSET_PATH})
+        LIST(APPEND LIBARCHIVE_LIBS_PRIVATE ${LIBCHARSET_PATH})
       ENDIF(HAVE_LOCALE_CHARSET)
     ENDIF(LIBCHARSET_PATH)
   ENDIF(LIBICONV_PATH)
@@ -1217,6 +1242,7 @@
   CMAKE_PUSH_CHECK_STATE()	# Save the state of the variables
   INCLUDE_DIRECTORIES(${LIBXML2_INCLUDE_DIR})
   LIST(APPEND ADDITIONAL_LIBS ${LIBXML2_LIBRARIES})
+  STRING(APPEND LIBSREQUIRED " libxml-2.0")
   SET(HAVE_LIBXML2 1)
   # libxml2's include files use iconv.h
   SET(CMAKE_REQUIRED_INCLUDES ${ICONV_INCLUDE_DIR} ${LIBXML2_INCLUDE_DIR})
@@ -1255,6 +1281,7 @@
       SET(HAVE_LIBXMLLITE 1)
       SET(XMLLITE_FOUND TRUE)
       LIST(APPEND ADDITIONAL_LIBS "xmllite" "uuid")
+      LIST(APPEND LIBARCHIVE_LIBS_PRIVATE xmllite uuid)
     ENDIF(HAVE_XMLLITE_H)
   ENDIF(WIN32 AND ENABLE_WIN32_XMLLITE)
   IF(NOT XMLLITE_FOUND)
@@ -1271,6 +1298,7 @@
       INCLUDE_DIRECTORIES(${EXPAT_INCLUDE_DIRS})
       LIST(APPEND CMAKE_REQUIRED_INCLUDES ${EXPAT_INCLUDE_DIR})
       LIST(APPEND ADDITIONAL_LIBS ${EXPAT_LIBRARIES})
+      STRING(APPEND LIBSREQUIRED " expat")
       SET(HAVE_LIBEXPAT 1)
       LA_CHECK_INCLUDE_FILE("expat.h" HAVE_EXPAT_H)
       CMAKE_POP_CHECK_STATE()	# Restore the state of the variables
@@ -1317,6 +1345,7 @@
         CHECK_FUNCTION_EXISTS_GLIBC(regcomp HAVE_REGCOMP_LIBREGEX)
         IF(HAVE_REGCOMP_LIBREGEX)
           LIST(APPEND ADDITIONAL_LIBS ${REGEX_LIBRARY})
+          LIST(APPEND LIBARCHIVE_LIBS_PRIVATE ${REGEX_LIBRARY})
           #
           # If regex.h is not found, retry looking for regex.h at
           # REGEX_INCLUDE_DIR
@@ -1366,6 +1395,7 @@
   IF(PCREPOSIX_FOUND)
     INCLUDE_DIRECTORIES(${PCRE_INCLUDE_DIR})
     LIST(APPEND ADDITIONAL_LIBS ${PCREPOSIX_LIBRARIES})
+    LIST(APPEND LIBARCHIVE_LIBS_PRIVATE  ${PCREPOSIX_LIBRARIES})
     # Test if a macro is needed for the library.
     TRY_MACRO_FOR_LIBRARY(
       "${PCRE_INCLUDE_DIR}" "${PCREPOSIX_LIBRARIES}"
@@ -1377,6 +1407,7 @@
 	ELSEIF(NOT WITHOUT_PCRE_STATIC AND NOT PCRE_STATIC AND PCRE_FOUND)
 	  # Determine if pcre static libraries are to be used.
       LIST(APPEND ADDITIONAL_LIBS ${PCRE_LIBRARIES})
+      LIST(APPEND LIBARCHIVE_LIBS_PRIVATE ${PCRE_LIBRARIES})
       SET(TMP_LIBRARIES ${PCREPOSIX_LIBRARIES} ${PCRE_LIBRARIES})
       MESSAGE(STATUS "trying again with -lpcre included")
       TRY_MACRO_FOR_LIBRARY(
@@ -1392,6 +1423,7 @@
         # ___chkstk_ms.
         MESSAGE(STATUS "Visual Studio build detected, trying again with -lgcc included")
         LIST(APPEND ADDITIONAL_LIBS ${LIBGCC_LIBRARIES})
+        LIST(APPEND LIBARCHIVE_LIBS_PRIVATE ${LIBGCC_LIBRARIES})
         SET(TMP_LIBRARIES ${PCREPOSIX_LIBRARIES} ${PCRE_LIBRARIES} ${LIBGCC_LIBRARIES})
           TRY_MACRO_FOR_LIBRARY(
             "${PCRE_INCLUDE_DIR}" "${TMP_LIBRARIES}"
@@ -1986,6 +2018,7 @@
     SET(CMAKE_REQUIRED_LIBRARIES "acl")
     FIND_LIBRARY(ACL_LIBRARY NAMES acl)
     LIST(APPEND ADDITIONAL_LIBS ${ACL_LIBRARY})
+    LIST(APPEND LIBARCHIVE_LIBS_PRIVATE ${ACL_LIBRARY})
 	  pkg_check_modules(LIBACL libacl)
 	  IF(LIBACL_VERSION)
 	    SET(LIBACL_PKGCONFIG_VERSION ${LIBACL_VERSION})
@@ -2131,6 +2164,7 @@
     SET(CMAKE_REQUIRED_LIBRARIES "richacl")
     FIND_LIBRARY(RICHACL_LIBRARY NAMES richacl)
     LIST(APPEND ADDITIONAL_LIBS ${RICHACL_LIBRARY})
+    LIST(APPEND LIBARCHIVE_LIBS_PRIVATE ${RICHACL_LIBRARY})
 	  pkg_check_modules(LIBRICHACL librichacl)
 	  IF(LIBRICHACL_VERSION)
 	    SET(LIBRICHACL_PKGCONFIG_VERSION ${LIBRICHACL_VERSION})
--- a/build/cmake/CreatePkgConfigFile.cmake
+++ b/build/cmake/CreatePkgConfigFile.cmake
@@ -8,7 +8,7 @@
 SET(includedir \${prefix}/include)
 # Now, this is not particularly pretty, nor is it terribly accurate...
 # Loop over all our additional libs
-FOREACH(mylib ${ADDITIONAL_LIBS})
+FOREACH(mylib ${LIBARCHIVE_LIBS_PRIVATE})
 	# Extract the filename from the absolute path
 	GET_FILENAME_COMPONENT(mylib_name ${mylib} NAME_WE)
 	# Strip the lib prefix
@@ -16,10 +16,6 @@
 	# Append it to our LIBS string
 	SET(LIBS "${LIBS} -l${mylib_name}")
 ENDFOREACH()
-# libxml2 is easier, since it's already using pkg-config
-FOREACH(mylib ${PC_LIBXML_STATIC_LDFLAGS})
-	SET(LIBS "${LIBS} ${mylib}")
-ENDFOREACH()
 # FIXME: The order of the libraries doesn't take dependencies into account,
 #	 thus there's a good chance it'll make some binutils versions unhappy...
 #	 This only affects Libs.private (looked up for static builds) though.
